### PR TITLE
refactor: move exps to `ErasedAst.IntrinsicOperator0`

### PR DESCRIPTION
**Pull request by jaschdoc**
Related to https://github.com/flix/flix/issues/5434

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/ast/ErasedAst.scala [hunk 1]
```diff
@@ -110,10 +106,14 @@ object ErasedAst {
 
   object IntrinsicOperator0 {
 
+    case object Region extends IntrinsicOperator0
+
     case object RecordEmpty extends IntrinsicOperator0
 
     case class GetStaticField(field: Field) extends IntrinsicOperator0
 
+    case class NewObject(name: String, clazz: java.lang.Class[_], methods: List[ErasedAst.JvmMethod]) extends IntrinsicOperator0
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/ast/ErasedAst.scala:115]: Do we know what name is? Is it the same as the name of the class?

Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/language/ast/ErasedAst.scala:122]: Yes it's used in JvmName: https://github.com/flix/flix/blob/25a47b7acb59c6bf3abf23e0e59566ec8c152c9c/main/src/ca/uwaterloo/flix/language/phase/jvm/JvmName.scala#L130

File: main/src/ca/uwaterloo/flix/language/phase/jvm/JvmOps.scala [hunk 2]
```diff
@@ -581,12 +576,26 @@ object JvmOps {
...
+
+        case IntrinsicOperator0.Cst(_) => Set.empty
+
+        case IntrinsicOperator0.Region => Set.empty
+
+        case IntrinsicOperator0.RecordEmpty => Set.empty
+
+        case IntrinsicOperator0.GetStaticField(_) => Set.empty
+
+        case IntrinsicOperator0.NewObject(_, _, methods) =>
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/jvm/JvmOps.scala:589]: I take it that this is necessary for the code to compile?

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/jvm/JvmOps.scala:589]: (which kind of breaks the idea that we should not have to match in each intrinsic)

**Review by magnus-madsen — COMMENTED** [review]
I suggest to move `NewObject` back into `Expression` -- for now.

File: main/src/ca/uwaterloo/flix/language/phase/jvm/JvmOps.scala (hunk 2 above)
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/jvm/JvmOps.scala:589]: I might be able to pre-compute `ClosureInfo` thus solving this.

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/jvm/JvmOps.scala:589]: I was able to precompute closure info

**Comment by magnus-madsen** [comment]
Yeah, we still have to keep NewObject in expression, at least for the moment.

**Comment by jaschdoc** [comment]
RFR

**Pull request merged**